### DISCUSSION TITLE
CasAdi support in python in the current version (2.6.19)

### DISCUSSION THREAD

**Discussion by joao-pm-santos96** [stack-of-tasks/pinocchio#2019] [Q&A] [answered]
Hello!
This package continues to amaze me, thank so much to the great developers being it!

After reading as much as I could I wasn't able to understand if there is or is not support for CasAdi in the python bindings at version 2.6.19. Can you please tell me if there is or isn't? I've tried `import pinocchio.casadi as cpin` but I receive a Module Not Found Error... but maybe I need to compile something on my side (?)

**Comment by nim65s** [comment, marked as answer]
Hi,

The current version is not really usable with casadi. Everything you need for this will be released in pinocchio v3.

Meanwhile, I can share a previous comment on this topic by @stephane-caron if you want a preview of pinocchio v3: https://github.com/stack-of-tasks/pinocchio/issues/1356#issuecomment-1641708731

**Reply by joao-pm-santos96** [reply]
This is awesome! Thank you!

**Reply by joao-pm-santos96** [reply]
Just one more little question. If I want to build poinocchio with CasADi by my own, should I do it in the devel branch or the pinocchio3-preview?

**Reply by nim65s** [reply]
probably pinocchio3-preview, but this probably won't be working as well as the binaries from jnrh2023
